hack/git-version.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeCarrier Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Check if the current commit has a matching tag.
+COMMIT=`git rev-parse HEAD`
+
+# git tag
+TAG=$(git describe --exact-match --abbrev=0 --tags ${COMMIT} 2> /dev/null || true)
+
+SUFFIX=''
+DETAIL=''
+
+if [ -z "$TAG" ]; then
+  # Use the last tag as base and create a new "dynamic" version from it.
+  TAG=$(git describe --abbrev=0)
+  COMMIT_COUNT=$(git --no-pager log ${TAG}..HEAD --oneline | wc -l)
+  COMMIT_COUNT_PADDED=$(printf %03d $COMMIT_COUNT)
+  SHORT_COMMIT_ID=$(git rev-parse --short HEAD)
```

Comment:
COMMIT_SHORT_SHA perhaps? When I'm defining some env vars I try to stay as close as possible to how CI systems usually define them, thus having naming consistency. Gitlab is pretty good in that regard: https://docs.gitlab.com/ee/ci/variables/